pkg/controllers/job/plugins/distributed-framework/mpi/mpi.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mpi
+
+import (
+	"flag"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	batch "volcano.sh/apis/pkg/apis/batch/v1alpha1"
+	pluginsinterface "volcano.sh/volcano/pkg/controllers/job/plugins/interface"
+)
+
+const (
+	MPIPluignName = "mpi"
+	defaultPort   = 22
+)
+
+type MpiPlugin struct {
+	MpiArguments []string
+	clientset    pluginsinterface.PluginClientset
+	MasterName   string
+	WorkerName   string
+	port         int
+}
+
+// New creates mpi plugin.
+func New(client pluginsinterface.PluginClientset, arguments []string) pluginsinterface.PluginInterface {
+	mp := MpiPlugin{MpiArguments: arguments, clientset: client}
+	mp.AddFlags()
+	return &mp
+}
+
+func (mp *MpiPlugin) AddFlags() {
+	flagSet := flag.NewFlagSet(mp.Name(), flag.ContinueOnError)
+	flagSet.StringVar(&mp.MasterName, "master", "master", "name of master role task")
+	flagSet.StringVar(&mp.WorkerName, "worker", "worker", "name of worker role task")
+	flagSet.IntVar(&mp.port, "port", defaultPort, "open port for containers")
+	if err := flagSet.Parse(mp.MpiArguments); err != nil {
+		klog.Errorf("plugin %s flagset parse failed, err: %v", mp.Name(), err)
+	}
+}
+
+func (mp *MpiPlugin) Name() string {
+	return MPIPluignName
+}
+
+func (mp *MpiPlugin) OnPodCreate(pod *v1.Pod, job *batch.Job) error {
+	// open port for ssh
+	for index, c := range pod.Spec.Containers {
+		SSHPortRight := false
+		for _, p := range c.Ports {
+			if p.ContainerPort == int32(mp.port) {
+				SSHPortRight = true
+				break
+			}
+		}
+		if !SSHPortRight {
+			sshPort := v1.ContainerPort{
+				Name:          "mpijob-port",
+				ContainerPort: int32(mp.port),
+			}
+			pod.Spec.Containers[index].Ports = append(pod.Spec.Containers[index].Ports, sshPort)
+		}
+	}
```

Comment:
Should we open ssh port for initContainers?

Reply:
ssh plugin mounted RSA keys?

Reply:
I'm not sure about all details MPI works. According to the [design doc](https://github.com/volcano-sh/volcano/blob/master/docs/user-guide/how_to_use_ssh_plugin.md), The abilities of `ssh` and `svc` are necessary for MPI jobs. If that, I guess the existing logic of these plugins should be added here, too.